### PR TITLE
Add image option in Values file

### PR DESCRIPTION
**Pull request by pratik141**
Add image option in Values Yaml file init container #506

### CODE REVIEW

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -833,6 +833,15 @@ image:
     # The policy that specifies when the user wants the opentelemetry collector images to be pulled
     pullPolicy: IfNotPresent
 
+  UniversalBaseImage:
+    # The registry and name of the Universal Base Image 8 image to pull
+    repository: registry.access.redhat.com/ubi8/ubi
+    # The tag of the Splunk OTel Collector image, default value is the chart appVersion
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/values.yaml:839]: this should be latest

File: helm-charts/splunk-otel-collector/values.yaml [hunk 2]
```diff
@@ -833,6 +833,15 @@ image:
...
+    tag: ""
+    # The policy that specifies when the user wants the opentelemetry collector images to be pulled
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/values.yaml:841]: this comment needs to be updated as well

File: helm-charts/splunk-otel-collector/values.yaml [hunk 3]
```diff
@@ -833,6 +833,14 @@ image:
     # The policy that specifies when the user wants the opentelemetry collector images to be pulled
     pullPolicy: IfNotPresent
 
+  UniversalBaseImage:
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/values.yaml:836]:
```suggestion
  universalBaseImage:
```

File: helm-charts/splunk-otel-collector/values.yaml [hunk 4]
```diff
@@ -833,6 +833,14 @@ image:
     # The policy that specifies when the user wants the opentelemetry collector images to be pulled
     pullPolicy: IfNotPresent
 
+  universalBaseImage:
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:836]: It doesn't have to UBI, can be any other linux image. So let's call it same as the container name for consistency with options in this section.
```suggestion
  # Image to be used by init container that patches log directories on the host, so the collector can read from them as a non-root user.
  # Effective only if `agent.securityContext.runAsUser` and `agent.securityContext.runAsGroup` are set to non-zero values.
  chown:
```

Comment (pratik141) [reply helm-charts/splunk-otel-collector/values.yaml]: updated

File: helm-charts/splunk-otel-collector/templates/_helpers.tpl [hunk 5]
```diff
@@ -209,6 +209,13 @@ Create the opentelemetry collector image name.
 {{- printf "%s:%s" .Values.image.otelcol.repository (.Values.image.otelcol.tag | default .Chart.AppVersion) -}}
 {{- end -}}
 
+{{/*
+Create the chown image name.
+*/}}
+{{- define "splunk-otel-collector.image.ubi" -}}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/_helpers.tpl:215]:
```suggestion
{{- define "splunk-otel-collector.image.chown" -}}
```

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 6]
```diff
@@ -136,7 +136,7 @@ spec:
         {{- end }}
         {{- if and $agent.securityContext.runAsUser $agent.securityContext.runAsGroup }}
         - name: chown
-          image: registry.access.redhat.com/ubi8/ubi
+          image: {{ template "splunk-otel-collector.image.ubi" . }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:139]:
```suggestion
          image: {{ template "splunk-otel-collector.image.chown" . }}
```

File: helm-charts/splunk-otel-collector/values.yaml [hunk 7]
```diff
@@ -833,6 +833,16 @@ image:
     # The policy that specifies when the user wants the opentelemetry collector images to be pulled
     pullPolicy: IfNotPresent
 
+    # Image to be used by init container that patches log directories on the host, so the collector can read from them as a non-root user.
+    # Effective only if `agent.securityContext.runAsUser` and `agent.securityContext.runAsGroup` are set to non-zero values.
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:837]:
```suggestion
  # Image to be used by init container that patches log directories on the host, so the collector can read from them as a non-root user.
  # Effective only if `agent.securityContext.runAsUser` and `agent.securityContext.runAsGroup` are set to non-zero values.
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**